service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/any/flight/update/UpdateControlledFlexibleResourceAttributesStep.java

```diff
@@ -0,0 +1,72 @@
+package bio.terra.workspace.service.resource.controlled.cloud.any.flight.update;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.UPDATE_FLEX_DATA;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.DbSerDes;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.resource.controlled.cloud.any.flexibleresource.ControlledFlexibleResource;
+import bio.terra.workspace.service.resource.controlled.cloud.any.flexibleresource.FlexibleResourceAttributes;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+
+public class UpdateControlledFlexibleResourceAttributesStep implements Step {
+  private final ResourceDao resourceDao;
+  private final ControlledFlexibleResource flexResource;
+
+  public UpdateControlledFlexibleResourceAttributesStep(
+      ResourceDao resourceDao, ControlledFlexibleResource flexResource) {
+    this.resourceDao = resourceDao;
+    this.flexResource = flexResource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    String previousAttributes = flexResource.attributesToJson();
+    context
+        .getWorkingMap()
+        .put(WorkspaceFlightMapKeys.ResourceKeys.PREVIOUS_ATTRIBUTES, previousAttributes);
+
+    final String updateData = context.getInputParameters().get(UPDATE_FLEX_DATA, String.class);
+
+    if (updateData == null) {
+      return StepResult.getStepResultSuccess();
+    }
+
+    String newAttributes =
+        DbSerDes.toJson(
+            new FlexibleResourceAttributes(
+                flexResource.getTypeNamespace(), flexResource.getType(), updateData));
+    boolean updated =
+        resourceDao.updateResource(
+            flexResource.getWorkspaceId(),
+            flexResource.getResourceId(),
+            null,
```

Comment:
nit: add comment for the null field so it's clear what they are. e.g. /*param=*/null